lib/components/narrative/narrative-itineraries-header.tsx

```diff
@@ -89,6 +102,16 @@ export default function NarrativeItinerariesHeader({
   const sortOptionsArr = sortOptions(intl, enabledSortModes)
   const sortText = sortOptionsArr.find((x) => x.value === sort.type)?.text
 
+  const mapExpansionText = useMemo(() => {
+    return mapExpanded
+      ? intl.formatMessage({
+          id: 'components.BatchResultsScreen.showResults'
+        })
+      : intl.formatMessage({
+          id: 'components.BatchResultsScreen.expandMap'
+        })
+  }, [intl, mapExpanded])
+
```

Comment:
Interesting, I think generally we would just create two strings and conditionally render them based on `mapExpanded` below, but this might be more efficient? I'll let the second reviewer decide